### ISSUE TITLE
Screen all black

### ISSUE THREAD

**Issue by feydreva** [SnosMe/awakened-poe-trade#185]
Hello, 

I run POE / Awakened trade on linux.
I have 2 screen.
1 for POE, and the other one.
When running awakened trade, it tales the second screen, but takes the all screen with a black page.
I have the interface when doind crtl-D, but the background covering the entire screen in black is still there.. and stay after..

Could it be possible to have in a windows instead of the whole screen
you can find a screen shot  here : https://imgur.com/a/TosI5TP

thank you

**Comment by Murdoc84** [comment]
I have 2 screen and no problem, do you run poe on main screen or second? I use main screen :)

**Comment by Storm-Engineer** [comment]
That happened to me when I tried it the first time, but then it went away.

However, APT will always open on my secondary screen instead of the primary where the game runs. The secondary is left of the Primary.

**Comment by Odzinic** [comment]
I'm having this issue on Manjaro KDE. Non-primary monitor is all black.

**Comment by weidtn** [comment]
I'm pretty sure that black window is the overlay. It is overlaying my game. I use xmonad and I am able to move the black overlay to my other screen with hotkeys.

I'm using Steam Proton-5.9-GE-6-ST and Vulkan (but it also happens with DirectX).

**Comment by Marcuss2** [comment]
I have a similar issue with a single monitor setup on KDE Plasma. I noticed compositing is disabled when I run Path of Exile.

I managed to bypass this issue by switching to Wayland, compositing is not disabled in this case and Awakened Poe Trade seems to work.

**Comment by SnosMe** [comment]
Enable window composition, because APT window needs transparency and spans entire size of PoE window.

**Comment by weidtn** [comment]
I enabled transparency in picom with `opacity-rule = ["10:name *?= 'Awakened PoE Trade'"]` it is still black. Not sure if that fixes the problem

**Comment by weidtn** [comment]
So this still is not fixed. Neither composition, nor disabling hardware acceleration do fix this. It does just show a black overlay both in KDE and Xmonad. The settings menu is visible and its possible to change transparency of it with the compositor

**Comment by botagar** [comment]
I faced this and what worked for me was switching the compositor renderer from OpenGL to XRender. Don't know if this helps too many people but it's worth a try.

**Comment by DarkOverlord69** [comment]
Hello, not sure if this is helpful, but I had this same black-screen issue too. I unchecked in System Settings -> Display and Monitor -> Compositor -> "Allow applications to block compositing". This works for me with both, OpenGL 2.0 and XRender.  I'm on Manjaro Linux too with KDE Plasma.
if more_information_is_needed == True: return givingMoreInfo()
cheers

**Comment by XMasterNinni** [comment]
Did anyone manage to get it running using picom? I'm experiencing the same problems (black box) using both xrender and glx.
It seems that it can be fixed using the KDE compositor and allowing composition blocking, but is there a picom equivalent to this setting?

Thanks and keep farming :)

**Comment by Dark32** [comment]
Try running with flags
`--enable-transparent-visuals --disable-gpu```
and with compton/picom

My compton config file:
```backend = "glx"
mark-wmwin-focused = true;
mark-ovredir-focused = true;
use-ewmh-active-win = false;
detect-rounded-corners = true;
detect-client-opacity = true;
refresh-rate = 0;
vsync = "none";
dbe = false;
paint-on-overlay = true;
unredir-if-possible = true;
focus-exclude = [ "class_g = 'Cairo-clock'" ];
detect-transient = true;
detect-client-leader = true;
invert-color-include = [ ];```
but I don't know what is options are strictly needed

**Comment by XMasterNinni** [comment]
Thanks! 
I can confirm that starting awakened-poe-trade with the  `--disable-gpu` flag did fix the problem for me using picom with glx backend. 

For anyone running a picom fork with background blurring on transparent windows enabled, note that you might need to disable blurring for awakened-poe-trade:

```
blur-background-exclude = [
  "class_g = 'awakened-poe-trade'",
   ...
];
```

**Comment by feydreva** [comment]
Hello, 

I running lastest Manjaro with KDE...

And I stil ahve the issue.. see screeshot bellow... no clack screen , but still not usable...

I tries with the --disable-gpu flags and the composition block flag in monitor parameter flag...

https://imgur.com/a/Ffuus0e

any advance on that issue ??

**Comment by warcayac** [comment]
> Hello, not sure if this is helpful, but I had this same black-screen issue too. I unchecked in **System Settings -> Display and Monitor -> Compositor -> "Allow applications to block compositing"**. This works for me with both, OpenGL 2.0 and XRender. I'm on Manjaro Linux too with KDE Plasma. `if more_information_is_needed == True: return givingMoreInfo()` cheers

This worked for me but when the popup window (prices window) tries to auto-hide itself, it does disappear but the focus is not returned the game, I have to switch windows to return the game. Any ideas?

**Comment by bruno-fs** [comment]
> This worked for me but when the popup window (prices window) tries to auto-hide itself, it does disappear but the focus is not returned to the game, I have to switch windows to return the game. Any ideas?

I'm having the same issue and I think this is the same described in the second item of [this](https://github.com/SnosMe/awakened-poe-trade/issues/162#issue-650847456). It seems there's a solution for it on discord. I'm digging there without luck :/

**Comment by UlverUA** [comment]
> This worked for me but when the popup window (prices window) tries to auto-hide itself, it does disappear but the focus is not returned to the game, I have to switch windows to return the game. Any ideas?
The same for me. Using Manjaro KDE. Anyone find the solution?

**Comment by flightlessmango** [comment]
> > This worked for me but when the popup window (prices window) tries to auto-hide itself, it does disappear but the focus is not returned to the game, I have to switch windows to return the game. Any ideas?
> 
> The same for me. Using Manjaro KDE. Anyone find the solution?

I ran into this issue as well with kde plasma and i3. I wrote a little hack to address it.
```
const { exec } = require('child_process');
exec('wmctrl -ia `wmctrl -l | grep "Path of Exile" | grep -v "Trade" | head -n1 | awk {"print $1"}`')
```
add this inside the `focusPoE` function in `overlay-window.ts` and make sure you have wmctrl

**Comment by nullagit** [comment]
Thank you to everyone in this thread, I managed to get Awakened POE Trade working with Devuan stable (Chimera) and DWM v6.3. Here are the steps I took. 

apt-get install picom
Create a config file (~/.config/picom.conf) with Dark32's settings
Add NINNiT's blur-background-exclude section at the bottom
Start picom
Open POE (I use steam w/Proton Experimental + Vulcan)
Open APT ($ ./Awakened-PoE-Trade-3.20.10003.AppImage --disable-gpu)

Works perfectly in DWM tiling mode, I have APT on my 2nd monitor and all is working perfectly!

**Comment by nullagit** [comment]
> add this inside the `focusPoE` function in `overlay-window.ts` and make sure you have wmctrl

Where is the overlay-window.ts file? I found an OverlayWindow.ts file in main/src/windowing/ but there's no `focusPoE` function in that file.

**Comment by SnosMe** [comment]
called `assertGameActive`  now

**Comment by baaa** [comment]
Apologies for resurrecting this many years after, but may be this'll help someone else.

I was using wayland with plasma on cachyos, and everything was working just fine. Even installing APT from packages via `yay -S awakened-poe-trade-bin` worked.

Then I switched to X11 and started having this "black screen in front of POE" error. Tried everything above, with all variations to no success.

What in the end worked for me was adding: `OpenGLIsUnsafe=false` flag to `~/.config/kwinrc` under `[Compositing]` section. which solved the issue.

I've reverted every other change I made (like using flags etc) and just the addition of `OpenGLIsUnsafe=false` solves it for me. So apparently Kwin marks some openGL features as unsafe, and that kills compositing for this case. May be this is an edge case just for my setup but who knows.
